### PR TITLE
make SRAM description value optional in SRAMInterface

### PR DESCRIPTION
**Pull request by debs-sifive**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- API modification

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

#### Release Notes

Add a `hasDescription` argument to `SRAMInterface` that specifies whether or not it contains a `SRAMDescription`.

Making `description` optional enables users to create `SRAMInterface`s without having to hook up an `SRAMDescription`.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/main/scala/chisel3/util/SRAM.scala [hunk 1]
```diff
@@ -575,21 +577,20 @@ object SRAM {
...
-    description.write := Property(numWritePorts)
-    description.readwrite := Property(numReadwritePorts)
-    description.maskGranularity := Property(
+    _out.description.get.depth := Property(size)
+    _out.description.get.dataWidth := Property(tpe.getWidth)
+    _out.description.get.masked := Property(isVecMem)
+    _out.description.get.read := Property(numReadPorts)
+    _out.description.get.write := Property(numWritePorts)
+    _out.description.get.readwrite := Property(numReadwritePorts)
+    _out.description.get.maskGranularity := Property(
```
Comment (jackkoenig) [inline src/main/scala/chisel3/util/SRAM.scala:586]:
```suggestion
    val description =  _out.description.get
    description.depth := Property(size)
    description.dataWidth := Property(tpe.getWidth)
    description.masked := Property(isVecMem)
    description.read := Property(numReadPorts)
    description.write := Property(numWritePorts)
    description.readwrite := Property(numReadwritePorts)
    description.maskGranularity := Property(
```
This diff can be simpler (and more elegant) if we stick with `val description`.

File: src/main/scala/chisel3/util/SRAM.scala [hunk 2]
```diff
@@ -575,21 +577,20 @@ object SRAM {
...
       Option
         .when(isVecMem)(tpe match {
           case t: Vec[_] => t.sample_element.getWidth
         })
         .getOrElse(0)
     )
-    description.hierarchy := Property(Path(mem))
+    _out.description.get.hierarchy := Property(Path(mem))
```
Comment (jackkoenig) [inline src/main/scala/chisel3/util/SRAM.scala:593]:
```suggestion
    description.hierarchy := Property(Path(mem))
```

File: src/test/scala/chiselTests/util/SRAMSpec.scala [hunk 3]
```diff
@@ -100,7 +100,7 @@ class SRAMSpec extends ChiselFlatSpec {
       val fooIo = IO(foo.cloneType)
       fooIo :<>= foo
     }
-    val chirrtl = emitCHIRRTL(new Top)
+    val chirrtl = emitCHIRRTL(new Top, Array("--full-stacktrace"))
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/util/SRAMSpec.scala:103]:
```suggestion
    val chirrtl = emitCHIRRTL(new Top)
```
No reason to commit that, that's just for debugging.

File: src/test/scala/chiselTests/util/SRAMSpec.scala [hunk 4]
```diff
@@ -242,11 +242,29 @@ class SRAMSpec extends ChiselFlatSpec {
...
+          numReadPorts = 0,
+          numWritePorts = 0,
+          numReadwritePorts = 1,
+          masked = false
+        )
+      )
+      sramIntf := DontCare
+    }
+    // should not give uninitialized errors for description properties
+    emitSystemVerilog(new Top)
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/util/SRAMSpec.scala:268]: Does `DontCare` affect Properties? Would this test pass even if `hasDescription` were `true`?

Comment (debs-sifive) [reply src/test/scala/chiselTests/util/SRAMSpec.scala:268]: As it currently stands, `DontCare` does not appear to affect properties. Is this likely to change in the future? I should rewrite the test if so...

> Would this test pass even if hasDescription were true?

The test fails if `hasDescription` is set to `true`

Comment (jackkoenig) [reply src/test/scala/chiselTests/util/SRAMSpec.scala:268]: I am not sure if it will ever change, probably not so I think we're okay.

**Pull request merged**